example/ExampleDialect.cpp

```diff
@@ -26,5 +26,28 @@
 #include "ExampleDialect.h"
 
 #define GET_INCLUDES
+#include "ExampleDialect.cpp.inc"
+
+const char *xd::toString(VectorKind kind) {
+  switch (kind) {
+  case VectorKind::LittleEndian:
+    return "LittleEndian";
+  case VectorKind::BigEndian:
+    return "BigEndian";
+  case VectorKind::MiddleEndian:
+    return "MiddleEndian";
+  default:
+    return "(bad)";
+  }
+}
+
+template <> struct llvm_dialects::Printable<xd::VectorKind> {
+  xd::VectorKind x;
+
+  explicit Printable(xd::VectorKind x) : x(x) {}
+
+  void print(llvm::raw_ostream &out) const { out << toString(x); }
+};
```

Comment:
I did try, but clearly didn't get it right the first time :)

I looked at it again, and now know why my first attempt failed: the operator<< has to be defined in the right namespace, otherwise it isn't included in the overload set that is used for overload resolution. ADL lookup FTW!